gridsome/app/directives/image.js

```diff
@@ -68,19 +68,9 @@ function loadImage (el) {
     return // src is already switched
   }
 
-  const image = new Image()
-
-  image.onload = function () {
-    el.classList.remove('g-image--loading')
-    el.classList.add('g-image--loaded')
-    
-    el.src = src
-    el.sizes = sizes
-    el.srcset = srcset
-  }
-
-  image.src = src
-
-  el.classList.add('g-image--loading')
-  el.classList.remove('g-image--loaded')
+  el.srcset = srcset
+  el.sizes = sizes
+  el.src = src
+  el.classList.remove('g-image--loading')
+  el.classList.add('g-image--loaded')
```

Comment:
I think it would be better to put this into the onload function. Also don't forget to declare the onload function before you assign the src to the element so the onload function is also called when the image is cached already. 😊

Something like

```js
el.onload = function () {
  el.classList.add('g-image--loaded')
}
el.srcset = srcset
el.src = src
```